### DISCUSSION TITLE
How to change order of statements?

### DISCUSSION THREAD

**Discussion by dekiesel** [pgmodeler/pgmodeler#1857] [Q&A] [answered]
For the purpose of validating emails I would like to

1. Install the extension `citext`
2. Create the type `citext`
3. Create a domain that uses the type `citext`

I have added the extension:
![image](https://github.com/pgmodeler/pgmodeler/assets/17333827/3662ea1b-881c-41c5-8972-50a3e4e802be)

Which has created the type:

![image](https://github.com/pgmodeler/pgmodeler/assets/17333827/0182ba4f-e357-4af9-bb5b-78eb1d2032d2)

I have created a domain that uses citext:
![image](https://github.com/pgmodeler/pgmodeler/assets/17333827/3559073b-bb1a-45a6-a83d-fc1eaeb979a5)

But when I export the model and try to load it in postgres I get:
```
psql:ddls/0_1_0.sql:1141: ERROR:  type "public.citext" does not exist
psql:ddls/0_1_0.sql:1141: STATEMENT:  CREATE DOMAIN public.domain_email AS public.citext
	CONSTRAINT check_mail CHECK (  VALUE ~ '^[a-zA-Z0-9.!#$%&''*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$');


```
When I grep for citext in the export:
```
grep citext ddls/0_1_0.sql 
CREATE DOMAIN public.domain_email AS public.citext
-- object: citext | type: EXTENSION --
-- DROP EXTENSION IF EXISTS citext CASCADE;
CREATE EXTENSION citext
-- -- object: public.citext | type: TYPE --
-- -- DROP TYPE IF EXISTS public.citext CASCADE;
-- CREATE TYPE public.citext AS

```
which shows that the domain is created first and that fails of course.

How can I influence the order of objects in the export?

**Comment by dekiesel** [comment]
When I created the objects pgmodeler complained that some objects where used before their definition. I clicked on "Apply fixes" and that made the validation pass, but the order is still wrong in the file.

**Reply by dekiesel** [reply]
Manually changing the order of the two statements makes the import succeed.

**Reply by rkhaotix** [reply, marked as answer]
I have to investigate, it can be a bug... Anyway, you can always change the object's creation order even if model validation fails to properly set up the desired order (there are cases that validation can't detect)!

https://pgmodeler.io/support/docs/67-model-validation?v=1.1.0

https://pgmodeler.io/support/docs/68-swap-objects-ids?v=1.1.0

**Reply by dekiesel** [reply]
Thank you, I'll have a look.

Let me know if I should send you my model.
